### PR TITLE
samples: add shell

### PR DESCRIPTION
**Pull request by wilu-nordic**

### CODE REVIEW

**Review by annakielar — CHANGES_REQUESTED** [review]

**Review by annakielar — APPROVED** [review]

File: samples/shell/sample.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+sample:
+  description: SHELL sample
+  name: SHELL
+
+common:
+    tags: pmic
+    timeout: 30
+
+tests:
+  sample.shell_example_eng_b:
```
Comment (masz-nordic) [inline samples/shell/sample.yaml:10]:
```suggestion
  sample.shell.eng_b:
```

File: samples/shell/README.rst [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+The sample also requires an nPM1300 EK.
+
+Overview
+********
+
+This sample do the following:
+
+* Configure the Zephyr shell driver.
+* Initialize and register a POF callback.
```
Comment (masz-nordic) [inline samples/shell/README.rst:27]:
```suggestion
This sample does the following:

* Configures the Zephyr shell driver.
* Initializes and registers a POF callback.
```
What do you think @annakielar ?

**Review by mkleszcz-no — APPROVED** [review]

File: samples/shell/README.rst (hunk 2 above)
Comment (annakielar) [reply samples/shell/README.rst:27]: Ohhh my bad. Definitely yes.

Comment (wilu-nordic) [reply samples/shell/README.rst]: FIxed

File: samples/shell/README.rst [hunk 3]
```diff
@@ -0,0 +1,95 @@
+.. _shell_sample:
+
+SHELL
+#####
+
+.. contents::
+   :local:
+   :depth: 2
+
+The SHELL sample allows to use the Zephyr shell to control a PMIC device.
```
[the remark this answers is not in the document]
Comment (wilu-nordic) [reply samples/shell/README.rst:10]: Zephyr Shell driver

**Review by annakielar — APPROVED** [review]

File: samples/shell/README.rst [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+The sample also requires an nPM1300 EK.
+
+Overview
+********
+
+This sample uses the Shell driver to do the following:
+
+* Configure the Zephyr Shell driver.
+* Initialize and register a POF callback.
```
[the remark this answers is not in the document]
Comment (wilu-nordic) [reply samples/shell/README.rst:27]: What about `The Shell sample allows to control a PMIC device with the Zephyr Shell driver.`?

File: samples/shell/README.rst [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+The sample supports the following development kits:
+
+.. table-from-sample-yaml::
+
+The sample also requires an nPM1300 EK.
+
+Overview
+********
+
+This sample does the following:
```
Comment (annakielar) [inline samples/shell/README.rst:24]:
```suggestion
This sample allows to do the following:
```

**Review by annakielar — CHANGES_REQUESTED** [review]

File: samples/shell/README.rst [hunk 6]
```diff
@@ -0,0 +1,95 @@
+.. _shell_sample:
+
+Shell
+#####
+
+.. contents::
+   :local:
+   :depth: 2
+
+The Shell sample allows to control a PMIC device with command set using the Zephyr shell module.
```
[the remark this answers is not in the document]
Comment (wilu-nordic) [reply samples/shell/README.rst:10]: Yes

File: samples/shell/README.rst [hunk 7]
```diff
@@ -0,0 +1,95 @@
+.. _shell_sample:
+
+Shell
+#####
+
+.. contents::
+   :local:
+   :depth: 2
+
+The Shell sample allows to control a PMIC device with a command set and using the Zephyr shell module.
```
Comment (annakielar) [inline samples/shell/README.rst:10]:
```suggestion
The Shell sample allows to control a PMIC device using a command set and the Zephyr shell module.
```

**Review by annakielar — APPROVED** [review]

File: samples/shell/README.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (masz-nordic) [reply samples/shell/README.rst]: No, the command set is built using the shell module.

File: samples/shell/README.rst [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+.. include:: /includes/build_and_run.txt
+
+Testing
+=======
+
+|test_sample|
+
+#. |connect_kit|
+#. |connect_terminal|
```
Comment (annakielar) [inline samples/shell/README.rst:90]: Any sample-specific steps?

File: samples/shell/README.rst [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+#. Review and use npmx commands in the Shell.
```
Comment (annakielar) [inline samples/shell/README.rst:91]:
```suggestion
#. Review available npmx commands, and use the relevant ones in the shell.
```

**Review by masz-nordic — APPROVED** [review]

**Review by annakielar — APPROVED** [review]

**Pull request merged**